packages/calling-component-bindings/src/utils/videoGalleryUtils.ts

```diff
@@ -85,22 +85,21 @@ export const convertRemoteParticipantToVideoGalleryRemoteParticipant = (
   let videoStream: VideoGalleryStream | undefined = undefined;
   let screenShareStream: VideoGalleryStream | undefined = undefined;
 
-  if (rawVideoStreamsArray[0]) {
-    if (rawVideoStreamsArray[0].mediaStreamType === 'Video') {
-      videoStream = convertRemoteVideoStreamToVideoGalleryStream(rawVideoStreamsArray[0]);
-    } else {
-      screenShareStream = convertRemoteVideoStreamToVideoGalleryStream(rawVideoStreamsArray[0]);
-    }
-  }
+  const sdkRemoteVideoStream =
+    Object.values(rawVideoStreamsArray).find((i) => i.mediaStreamType === 'Video' && i.isAvailable) ||
+    Object.values(rawVideoStreamsArray).find((i) => i.mediaStreamType === 'Video');
 
-  if (rawVideoStreamsArray[1]) {
-    if (rawVideoStreamsArray[1].mediaStreamType === 'ScreenSharing') {
-      screenShareStream = convertRemoteVideoStreamToVideoGalleryStream(rawVideoStreamsArray[1]);
-    } else {
-      videoStream = convertRemoteVideoStreamToVideoGalleryStream(rawVideoStreamsArray[1]);
-    }
+  const sdkScreenShareStream =
+    Object.values(rawVideoStreamsArray).find((i) => i.mediaStreamType === 'ScreenSharing' && i.isAvailable) ||
+    Object.values(rawVideoStreamsArray).find((i) => i.mediaStreamType === 'ScreenSharing');
+
+  if (!sdkRemoteVideoStream || !sdkScreenShareStream) {
+    throw new Error('Video stream or screen share stream is undefined');
```

Comment:
Why do they both have to be defined here? I'f i"m not screensharing won't sdkScreenShareStream be undefined?

Reply:
Every participant joins the call will always have at least 2 streams

Even you don't turn anything on, they will be available=false

Also it will cause much worse problem from previous logic - we direct visit streamArray[1] index of the stream array, since we don't get any serve bug for now, we can confirm there are always at least 2 streams in the array

Reply:
Hmm, double checked the logic, looks like there is a check for undefined of previous logic, lemme fix this